analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -516,59 +516,64 @@ public void DeriveFromTrackedCollection()
+        list.Clear();                                       // Noncompliant should raise, because the methods above should not reset the state
+
         list.Cast<byte>();                                  // FN
-        list.Concat(list);                                  // FN
-        list.Contains(5, EqualityComparer<int>.Default);    // FN
-        list.Count();                                       // FN
-        list.DefaultIfEmpty();                              // FN
-        list.Distinct();                                    // FN
-        list.Except(list);                                  // FN
-        list.First();                                       // FN
-        list.FirstOrDefault();                              // FN
-        list.GroupBy(x => x);
-        list.GroupJoin(list, x => x, x => x, (x, y) => x);  // FN
-        list.Intersect(list);                               // FN
-        list.Join(list, x => x, x => x, (x, y) => x);       // FN
-        list.Last();                                        // FN
-        list.LastOrDefault();                               // FN
-        list.LongCount();                                   // FN
-        list.Max();                                         // FN
-        list.Min();                                         // FN
         list.OfType<int>();                                 // FN
-        list.OrderBy(x => x);
-        list.OrderByDescending(x => x);
-        list.Select(x => x);                                // FN
-        list.SelectMany(x => new int[5]);                   // FN
-        list.SequenceEqual(list);                           // FN
-        list.Single();                                      // FN
-        list.SingleOrDefault();                             // FN
-        list.Skip(1);                                       // FN
-        list.SkipWhile(x => true);                          // FN
-        list.Sum();                                         // FN
-        list.Take(1);                                       // FN
-        list.TakeWhile(x => true);                          // FN
-        list.ToArray();                                     // FN
-        list.ToDictionary(x => x);                          // FN
-        list.ToList();                                      // FN
-        list.ToLookup(x => x);
-        list.Union(list);                                   // FN
-        list.Where(x => true);                              // FN
-        list.Zip(list, (x, y) => x);                        // FN
-        Enumerable.Reverse(list);                           // FN
-        list.Clear();                                       // FN, should raise, because the methods above should not reset the state
+        list.AsEnumerable();
+        list.AsQueryable();
+        list.AsReadOnly();                               
+        list.Concat(list);
+        list.Count();
+        list.DefaultIfEmpty();
```

Comment:
Why is not `Count` and everything below also supported? Or why was the `// FN` comment removed from them?

Reply:
IMO all of the following should be compliant, but I'm open to discussion.
```cs
list.AsEnumerable();
list.AsQueryable();
list.AsReadOnly();                               
list.Concat(list);
list.DefaultIfEmpty();
list.Join(list, x => x, x => x, (x, y) => x);
list.ToArray();                                    
list.ToDictionary(x => x);                         
list.ToList();                                     
list.ToLookup(x => x);                         
list.Union(list);
```

Reply:
Can you explain why?
For example 
```
var list = new List<int>();
var zero = list.DefaultIfEmpty();
```
is the same as
```
var zero = 0;
```